### DISCUSSION TITLE
[HELP] ERROR: function gen_random_uuid() does not exist

### DISCUSSION THREAD

**Discussion by fbonhomm** [prisma/prisma#6453] [Q&A]
Hello,
I have an error when I make a migration with as file:

```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model task {
  id         String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  created_at DateTime @default(now()) @db.Timestamptz(6)
  updated_at DateTime @default(now()) @db.Timestamptz(6)
  name       String
  done       Boolean  @default(false)
}
```

And the error:
```
Error: Database error
Error querying the database: db error: ERROR: function gen_random_uuid() does not exist
HINT: No function matches the given name and argument types. You might need to add explicit type casts.
   0: migration_core::api::ApplyMigrations
             at migration-engine/core/src/api.rs:73
```

Do you a hint for resolve this error ?

Thank you

PS: Prisma version `2.20.1`

**Comment by ryands17** [comment]
@fbonhomm 👋 

If you're using Prisma Migrate, you need to add this to the migration `.sql` file for the function to exist:

```sql
CREATE EXTENSION pgcrypto;
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
